src/runner/PlutoRunner.jl

```diff
@@ -176,7 +176,7 @@ replace_pluto_properties_in_expr(other; kwargs...) = other
 "Similar to [`replace_pluto_properties_in_expr`](@ref), but just checks for existance and doesn't check for [`GiveMeCellID`](@ref)"
 has_hook_style_pluto_properties_in_expr(::GiveMeRerunCellFunction) = true
 has_hook_style_pluto_properties_in_expr(::GiveMeRegisterCleanupFunction) = true
-has_hook_style_pluto_properties_in_expr(expr::Expr) = any(has_hook_style_pluto_properties_in_expr, expr.args)
+has_hook_style_pluto_properties_in_expr(expr::Expr)::Bool = any(has_hook_style_pluto_properties_in_expr, expr.args)
```

Comment:
In which way does adding type assertion help? `any` should be easy to infer.

Reply:
Normally, yes. But `expr.args` is a `Vector{Any}`:

```julia
julia> ex = :(1 + 1)
:(1 + 1)

julia> f(ex) = ex.args;

julia> @code_warntype f(ex)
MethodInstance for f(::Expr)
  from f(ex) in Main at REPL[15]:1
Arguments
  #self#::Core.Const(f)
  ex::Expr
Body::Vector{Any}
1 ─ %1 = Base.getproperty(ex, :args)::Vector{Any}
└──      return %1
```
which results in a `Union{Missing, Bool}` instead of `Bool`:
```julia
julia> g() = any(==("foo"), ex.args)

julia> @code_warntype g()
MethodInstance for g()
  from g() in Main at REPL[17]:1
Arguments
  #self#::Core.Const(g)
Body::Union{Missing, Bool}
1 ─ %1 = (==)("foo")::Base.Fix2{typeof(==), String}
│   %2 = Base.getproperty(Main.ex, :args)::Any
│   %3 = Main.any(%1, %2)::Union{Missing, Bool}
└──      return %3
```
In general, you're right indeed.
```julia
julia h() = any(==("foo"), ["bar"]);

```
has return type `Bool`.

Reply:
But does a return type help with inference? I thought that `f(x) = z(x)::Bool` helps, but `f(x)::Bool = z(x)` does not? Not sure why i think that

Reply:
@rikhuijzer

Reply:
Sorry for late reply. I didn't see this one yet.

> But does a return type help with inference? I thought that `f(x) = z(x)::Bool` helps, but `f(x)::Bool = z(x)` does not? Not sure why i think that

Depends on where you care about the inference. In this case, the assertion on the method for `has_hook_style_pluto_properties_in_expr` doesn't help because the `any` returns an `Union{Missing,Bool}` and the type annotation on the function sets that to `Bool`.

However, this may help a lot for places where the method is **used**, because thanks to the annotation, the type can be inferred which, in turn, may allow more code to be hit during precompilation.

In this case, I just added it because `has_hook_style_pluto_properties_in_expr` **must** be a boolean at line 281. So, adding the annotation improves:
1. code readability
2. code inferability
3. debugability because it fails at an earlier point. Namely, when an incorrect type is returned